src/auto-update-manager.js

```diff
@@ -0,0 +1,84 @@
+'use babel'
+
+import {Emitter, CompositeDisposable} from 'event-kit'
+import {ipcRenderer} from 'electron'
+
+export default class AutoUpdateManager {
+  constructor ({applicationDelegate}) {
+    this.subscriptions = new CompositeDisposable()
+    this.emitter = new Emitter()
+
+    this.subscriptions.add(
+      applicationDelegate.onDidBeginCheckingForUpdate(() => {
+        this.emitter.emit('did-begin-checking-for-update')
+      }),
+      applicationDelegate.onDidBeginDownloadingUpdate(() => {
+        this.emitter.emit('did-begin-downloading-update')
+      }),
+      applicationDelegate.onDidCompleteDownloadingUpdate((details) => {
+        this.emitter.emit('did-complete-downloading-update', details)
+      }),
+      applicationDelegate.onUpdateNotAvailable(() => {
+        this.emitter.emit('update-not-available')
+      })
+    )
+  }
+
+  dispose () {
```

Comment:
For consistency with all other classes in Atom, we should probably call this `destroy`. Disposables are intended to represent an indirect claim on a specific resource, whereas all our application objects such as the workspace, panes, editors, selections, etc implement a `destroy` method enabling them to be destroyed. Destruction is a simpler concept. This object exists, and now it shouldn't any longer.

It's a subtle difference and maybe not one worth maintaining, but for now I think we should be consistent. If we really want to change this, we should deprecate `destroy` everywhere and call it `dispose`.